src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const STATECOPY = {};
+
+  Object.assign(STATECOPY, state);
+
+  const modificationsHistory = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(STATECOPY, action.extraData);
+
+      const STATESARRAY = { ...STATECOPY };
+
+      modificationsHistory.push(STATESARRAY);
```

Comment:
You're creating a new copy of the state for each action. This is repetitive and could be extracted to after the switch statement to avoid duplicating code.